src/core/strvector.c

```diff
@@ -312,18 +294,18 @@ igraph_error_t igraph_strvector_append(igraph_strvector_t *to,
     igraph_bool_t error = 0;
     IGRAPH_CHECK(igraph_strvector_resize(to, len1 + len2));
     for (i = 0; i < len2; i++) {
-        if (from->data[i][0] != '\0') {
-            IGRAPH_FREE(to->data[len1 + i]);
-            to->data[len1 + i] = strdup(from->data[i]);
-            if (!to->data[len1 + i]) {
+        if (from->stor_begin[i][0] != '\0') {
+            IGRAPH_FREE(to->stor_begin[len1 + i]);
```

Comment:
I think the code is wrong and it should be fixed 🙂 I will use a reserve instead of resize.

After that fix, there's another thing wrong here. This only copies strings if they're not empty. That seems like a strange exception, which I will remove.